### PR TITLE
Category api

### PR DESCRIPTION
**Pull request by mrkiencom**

### CODE REVIEW

File: src/app.module.ts [hunk 1]
```diff
@@ -1,10 +1,31 @@
...
+    TypeOrmModule.forRoot({
+      type: 'postgres',
+      host: 'localhost',
+      port: 5432,
+      username: 'postgres',
+      password: 'postgres',
+      database: 'book-management',
+      autoLoadEntities: true,
+      synchronize: true,
+    }),
```
Comment (tranmanhlinh) [inline src/app.module.ts:36]: Add these values to env file.

File: src/auth/auth.controller.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+export class AuthController {
+  constructor(private authService: AuthService) {}
+  @Post('/signin')
+  signIn(@Body() account: AuthUser): Promise<{ accessToken: string }> {
+    return this.authService.signIn(account);
+  }
+  @Post('/test')
+  @UseGuards(AuthGuard())
+  test(@Req() req) {
+    console.log(req);
```
Comment (tranmanhlinh) [inline src/auth/auth.controller.ts]: Remove this.

File: src/auth/auth.service.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  ) {}
+  async signIn(account: AuthUser): Promise<{ accessToken: string }> {
+    const { email, password } = account;
+    const user = await this.userRepository.findOne({ email });
+    if (user && (await bcrypt.compareSync(password, user.password))) {
+      const accessToken: string = await this.jwtService.sign({ email });
+      return { accessToken };
+    } else {
+      console.log(user.password);
+      throw new UnauthorizedException('please check your loggin credentials');
```
Comment (tranmanhlinh) [inline src/auth/auth.service.ts]: Store all message value in a constant file.

File: src/auth/dto/auth-user.dto.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { IsNotEmpty } from 'class-validator';
+
+export class AuthUser {
+  @IsNotEmpty()
+  email: string;
+  @IsNotEmpty()
+  password: string;
```
Comment (tranmanhlinh) [inline src/auth/dto/auth-user.dto.ts:10]: Add empty lines between fields.

File: src/user/users.repository.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+      lastName,
+      avatar,
+    });
+    try {
+      return this.save(newUser);
+    } catch (error) {
+      console.log(error);
+      if (error.code === 23505) {
+        throw new ConflictException('Email a already exits');
+      } else throw new InternalServerErrorException();
```
Comment (tranmanhlinh) [inline src/user/users.repository.ts]: if () { throw A; }
throw B;

File: src/auth/auth.module.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+import { JwtStrategy } from './jwt.strategy';
+
+@Module({
+  imports: [
+    PassportModule.register({ defaultStrategy: 'jwt' }),
+    JwtModule.register({
+      secret: 'topSecret51',
+      signOptions: {
+        expiresIn: 3600,
+      },
```
Comment (tranmanhlinh) [inline src/auth/auth.module.ts]: Add it to env file.

File: src/app.module.ts (hunk 1 above)
Comment (mrkiencom) [reply src/app.module.ts:36]: done

File: src/auth/auth.controller.ts (hunk 2 above)
Comment (mrkiencom) [reply src/auth/auth.controller.ts]: done

File: src/auth/auth.service.ts (hunk 3 above)
Comment (mrkiencom) [reply src/auth/auth.service.ts]: done

File: src/auth/dto/auth-user.dto.ts (hunk 4 above)
Comment (mrkiencom) [reply src/auth/dto/auth-user.dto.ts:10]: done

File: src/user/users.repository.ts (hunk 5 above)
Comment (mrkiencom) [reply src/user/users.repository.ts]: done

File: src/auth/auth.module.ts (hunk 6 above)
Comment (mrkiencom) [reply src/auth/auth.module.ts]: done

File: src/user/user.controller.ts [hunk 7]
```diff
@@ -7,14 +7,20 @@ import {
...
 import { UserService } from './user.service';
 
 @Controller('user')
 export class UserController {
-  constructor(private userService: UserService) {}
+  constructor(
+    private userService: UserService,
+    private configService: ConfigService,
+  ) {
+    console.log(configService.get('TEST_VALUE'));
```
Comment (tranmanhlinh) [inline src/user/user.controller.ts]: Remove console.log

File: src/user/user.entity.ts [hunk 8]
```diff
@@ -8,9 +8,9 @@ export class User {
   @Column()
   password: string;
   @Column()
-  firstName: string;
+  first_Name: string;
   @Column()
-  lastName: string;
+  last_Name: string;
```
Comment (tranmanhlinh) [inline src/user/user.entity.ts]: Camelcase.
 @Column({name: last_name})
  lastName: string;

File: src/user/user.service.ts [hunk 9]
```diff
@@ -28,8 +28,8 @@ export class UserService {
       id: id,
       email: email,
       password: password,
-      firstName: firstName,
-      lastName: lastName,
+      first_Name: firstName,
+      last_Name: lastName,
```
Comment (tranmanhlinh) [inline src/user/user.service.ts]: Same here.

File: src/user/users.repository.ts [hunk 10]
```diff
@@ -18,8 +18,8 @@ export class UserRepository extends Repository<User> {
     const newUser = this.create({
       email,
       password: hashedPassword,
-      firstName,
-      lastName,
+      first_Name: firstName,
+      last_Name: lastName,
```
Comment (tranmanhlinh) [inline src/user/users.repository.ts:22]: Same

File: src/user/user.service.ts (hunk 9 above)
Comment (mrkiencom) [reply src/user/user.service.ts]: done

File: src/user/users.repository.ts (hunk 10 above)
Comment (mrkiencom) [reply src/user/users.repository.ts:22]: done

File: src/user/user.entity.ts (hunk 8 above)
Comment (mrkiencom) [reply src/user/user.entity.ts]: done

File: src/user/user.controller.ts (hunk 7 above)
Comment (mrkiencom) [reply src/user/user.controller.ts]: done

**Pull request merged**